src/vconsole/vconsole-setup.c

```diff
@@ -316,17 +276,47 @@ int main(int argc, char **argv) {
                         log_warning_errno(r, "Failed to read /proc/cmdline: %m");
         }
 
-        if (utf8)
-                (void) enable_utf8(fd);
-        else
-                (void) disable_utf8(fd);
+        /* Sanitize vc_cnt */
+        if (vc_cnt == NULL)
+                vc_max = 12;
+        else {
+                r = safe_atoi(vc_cnt, &vc_max);
+                if (r || vc_max < 1 || vc_max > MAX_CONSOLES) {
+                        log_error("Invalid value for VC_MAX (vconsole.vc.max), it should be between 1 and " STRINGIFY(MAX_CONSOLES));
+                        return EXIT_FAILURE;
```

Comment:
It's better to warn and continue here using the default... Setting too many / too few consoles is not a big issue.